Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -762,7 +762,7 @@ WebIDL::ExceptionOr<void> set_up_readable_stream_default_controller_from_underly
     if (underlying_source.start) {
         start_algorithm = [controller, underlying_source_value, callback = underlying_source.start]() -> WebIDL::ExceptionOr<JS::Value> {
             // Note: callback does not return a promise, so invoke_callback may return an abrupt completion
-            return WebIDL::invoke_callback(*callback, underlying_source_value, controller);
+            return TRY(WebIDL::invoke_callback(*callback, underlying_source_value, controller)).release_value();
```

Comment:
This seems like quite the footgun, at least to me. I would not expect `WebIDL::ExceptionOr` to require a `JS::Completion` to be errored. I was thinking of making that constructor `explicit`, or maybe a `WebIDL::from_errored_completion(JS::Completion)` to be even more clear. Thoughts?

Reply:
Hmm, how would that more explicit API look in practice? 🤔

Reply:
The idea would be to remove the implicit `WebIDL::ExceptionOr(JS::Completion)` constructor (or rather, make it private) and replace it with `WebIDL::ExceptionOr::from_errored_completion(JS::Completion)`. That way the code I had here originally wouldn't have compiled, and I would have instead had to write `WebIDL::ExceptionOr::from_errored_completion(WebIDL::invoke_callback(...))`. But that isn't the behavior I want, since the completion may not be errored, so I would have instead landed on what I wrote in this PR: `TRY(...).release_value()`. 

The other option would be to handle successful completions, but I'm assuming there is some reason that that situation was disallowed in the first place. If there isn't (i.e. it was just because it wasn't necessary at the time), maybe we could change the constructor to be something more like (note: note the final result):
```c++
ExceptionOr(JS::Completion result_or_exception)
    : m_result_or_exception(move(result_or_exception))
{
    auto const& completion = m_result_or_exception.template get<JS::Completion>();
    if constexpr (requires { ValueType(JS::Value); }) {
        m_result_or_completion = ValueType(m_result_or_completion.get<JS::Completion>()->release_value());
    } else {
        VERIFY(completion.is_error());
    }
}
```
That way a successful completion value would be stored in the `ValueType` branch of the `Variant` and treated as a non-error.

Reply:
I think an explicit API would be worse here, we would then lose out on the ability to `TRY` any operation that returns a `JS::Completion` or `JS::ThrowCompletionOr`, which we do all over the place.

I don't see an issue with `ExceptionOr` handling value vs. error completions.

Reply:
Ahh right. I'll look into the value support then.